### PR TITLE
Added ClickHouse Python driver integration

### PR DESCRIPTION
**Pull request by lesandie**
I have read the CLA Document and I hereby sign the CLA.

Compiled different use cases/environments/problems in a kb article

### CODE REVIEW

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+## ClickHouse python drivers
+
+There are two main python drivers that can be used with ClickHouse. They all have their different set of features and use cases:
+
+### ClickHouse driver AKA [clickhouse-driver](https://clickhouse-driver.readthedocs.io/en/latest/)
+
+The **`clickhouse-driver`** is a Python library used for interacting with ClickHouse. Here's a summary of its features:
+
+1. **Connectivity**: **`clickhouse-driver`** allows Python applications to connect to ClickHouse servers over TCP/IP Native Interface (9000/9440 ports) and also HTTP interface bu it is experimental.
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:17]: ...but it is experimental. (missing the t on "but")

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+2. **SQL Queries**: It enables executing SQL queries against ClickHouse databases from Python scripts, including data manipulation (insertion, deletion, updating) and data retrieval (select queries).
+3. **Query Parameters**: Supports parameterized queries, which helps in preventing SQL injection attacks and allows for more efficient execution of repeated queries with different parameter values.
+4. **Connection Pooling**: Provides support for connection pooling, which helps manage connections efficiently, especially in high-concurrency applications, by reusing existing connections instead of creating new ones for each query.
+5. **Data Types**: Handles conversion between Python data types and ClickHouse data types, ensuring compatibility and consistency when passing data between Python and ClickHouse.
+6. **Error Handling**: Offers comprehensive error handling mechanisms, including exceptions and error codes, to facilitate graceful error recovery and handling in Python applications.
+7. **Asynchronous Support**: Supports asynchronous execution of queries using asyncio, allowing for non-blocking query execution in asynchronous Python applications.
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:23]: Is it worth putting `asyncio` in backquotes? (same for other mentions in text.)

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+- Good Pandas/Numpy support: [https://clickhouse-driver.readthedocs.io/en/latest/features.html#numpy-pandas-support](https://clickhouse-driver.readthedocs.io/en/latest/features.html#numpy-pandas-support)
+- Good SQLALchemy support: [https://pypi.org/project/clickhouse-sqlalchemy/](https://pypi.org/project/clickhouse-sqlalchemy/)
+
+This was the first python driver for clickhouse. It has a mature codebase. By default clickhouse drivers uses [synchronous code](https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading). There is a wrapper to convert code to asynchronous, [https://github.com/long2ice/asynch](https://github.com/long2ice/asynch)
+
+Here you can get a basic working example from Altinity repo for ingestion/selection using clickhouse-driver:
+
+[https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_ch_driver.py](https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_ch_driver.py)
+
+### ClickHouse-connect AKA clickhouse-connect
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:39]: Make `clickhouse-connect` a link like the other library?

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    - [https://clickhouse.com/docs/en/integrations/python#managing-clickhouse-session-ids](https://clickhouse.com/docs/en/integrations/python#managing-clickhouse-session-ids)
+    - If you want to specify a session_id per query you should be able to use the setting dictionary to pass a `session_id` for each query (note that ClickHouse will automatically generate a `session_id` if none is provided).
+    
+    ```python
+    SETTINGS = {"session_id": "dagster-batch" + "-" + f"{time.time()}"}
+    client.query("INSERT INTO table ....", settings=SETTINGS)
+    ```
+    
+
+Also in clickhouse documentation some explanatio how to set session_id with another approach: [https://clickhouse.com/docs/en/integrations/python#managing-clickhouse-session-ids](https://clickhouse.com/docs/en/integrations/python#managing-clickhouse-session-ids)
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:93]: "explanation" missing an n, `session_id` in backticks for consistency

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+[ClickHouse Connect Driver API | ClickHouse Docs](https://clickhouse.com/docs/en/integrations/language-clients/python/driver-api#common-method-arguments)
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:95]: link should be to https://clickhouse.com/docs/en/integrations/python#clickhouse-connect-driver-api, I think.

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+# Run the async main function
+if __name__ == '__main__':
+    asyncio.run(main())
+```
+
+Clickhouse-driver code is also synchronous and suffers the same problem as clickhouse-connect
+
+[https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading](https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading)
+
+So to use asynchronous approach it is recommended to use a connection pool and some asyncio wrapper that can hide the complexity of using the `ThreadPoolExecutor/ProcessPoolExecutor`
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:140]: ...an asynchronous approach...

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+[https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_aiohttp_inserts.py](https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_aiohttp_inserts.py)
+
+This will use simply requests module and aiohttp (you can tune the connection pooler [https://docs.aiohttp.org/en/stable/client_advanced.html#limiting-connection-pool-size](https://docs.aiohttp.org/en/stable/client_advanced.html#limiting-connection-pool-size))
+
+Also `aiochclient` is another good wrapper  [https://github.com/maximdanilchenko/aiochclient](https://github.com/maximdanilchenko/aiochclient) for the HTTP interface
+
+For the native interface you can try [https://github.com/long2ice/asynch](https://github.com/long2ice/asynch)
+
+`asynch` is an asyncio ClickHouse Python Driver with native (TCP) interface support, which reuse most of [clickhouse-driver](https://github.com/mymarilyn/clickhouse-driver) and comply with [PEP249](https://www.python.org/dev/peps/pep-0249/).
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:152]: ...reuses most of...and complies with...

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+[https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading](https://clickhouse-driver.readthedocs.io/en/latest/quickstart.html#async-and-multithreading)
+
+So to use asynchronous approach it is recommended to use a connection pool and some asyncio wrapper that can hide the complexity of using the `ThreadPoolExecutor/ProcessPoolExecutor`
+
+To begin testing such environment [aiohttp](https://docs.aiohttp.org/) is a good approach. Here an example:
+
+[https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_aiohttp_inserts.py](https://github.com/lesandie/clickhouse-tests/blob/main/scripts/test_aiohttp_inserts.py)
+
+This will use simply requests module and aiohttp (you can tune the connection pooler [https://docs.aiohttp.org/en/stable/client_advanced.html#limiting-connection-pool-size](https://docs.aiohttp.org/en/stable/client_advanced.html#limiting-connection-pool-size))
```
Comment (DougTidwell) [inline content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:146]: not a python guy, but does this make sense? "This simply uses the `requests` module and `aiohttp` ...

**Review by DougTidwell — APPROVED** [review]
Looks great! There's one link that you might want to change; beyond that nothing but nits.

File: content/en/altinity-kb-integrations/ClickHouse_python_drivers.md (hunk 8 above)
Comment (lesandie) [reply content/en/altinity-kb-integrations/ClickHouse_python_drivers.md:146]: Yes not well expressed I'll rewrite

**Pull request merged**